### PR TITLE
feat: allow mtest to be installed

### LINKED ISSUE #2: Properly label mtest in CMake to allow external use
Allow mtest to be properly labelled in CMake with project exports to allow distribution and external use.

This would allow one to use `find_package(mtest)` and (possibly?) `pkg-config` with mtest such that a package manager can handle the install/update of mtest as a project.

This is compared to the current methods of installing mtest using `FetchContent` or git submodules. Neither allows an external management of dependencies, for example in packaging up C projects.

An example layout for this would be like:
- The `include` folder gets exported into a `include` or `include/mtest`.
- The `cmake` folder gets exported into a `share/cmake` or `share/cmake/mtest` folder.
- Set up the relevant package descriptions for versioning. (F.x. `COMPATIBILITY AnyNewerVersion`).

I have personally been looking at [how googletest does this](https://github.com/google/googletest/blob/main/googletest/CMakeLists.txt), but my skills with CMake is sadly not sufficient for creating a PR with these changes applied.

### PR DESCRIPTION
**Pull request by soupglasses**
Took me a bit, but its now possible to install `mtest` as a system level dependency, which allows the use of `find_package(mtest)`. :tada:

Tried my best to explain how this works in the comments, but it's sadly by nature quite complicated (because CMake). Input appreciated! 

Fixes: #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,35 +78,57 @@ FetchContent_Declare(mtest
...
+other approaches, as this does not ensure that anyone using your project will
+have the same mtest library installed as you, or even that its installed.
+
+However, this approach can be helpful if you are packaging your project, or have
+otherwise installed mtest as part of your operating system.
+
+If you do not have mtest already installed, you can run the following two
+commands:
+```bash
+cmake -B cmake_build -S . -DCMAKE_BUILD_TYPE=Release -DMTEST_INSTALL=ON
```
Comment (MortenSchou) [inline README.md]: Looks like the MTEST_INSTALL option is no longer used.
```suggestion
cmake -B cmake_build -S . -DCMAKE_BUILD_TYPE=Release
```

File: README.md [hunk 2]
```diff
@@ -27,35 +78,57 @@ FetchContent_Declare(mtest
...
+Updating mtest can be done with the `git submodule update` command, then
+using git to commit the updated submodule directory.
 
-TEST_CASE(my_other_test_case, {
-  double i = 3.14;
-  // When comparing doubles, we need to specify a tolerance - here we choose 0.01
-  REQUIRE_EQ_DOUBLE(4.0, i, 0.01);  // Test fails and the test case exits.
-  CHECK_TRUE(1);                    // Not executed but it would succeed.
-})
+### NOT RECOMMENDED: Using system installed `mtest`
```
Comment (MortenSchou) [inline README.md]: Maybe make text slightly less scary 😃 
```suggestion
### Using system installed `mtest` (not recommended)
```

**Comment by MortenSchou** [comment]
This looks good. 👍 
I tested that FetchContent still works.

File: README.md (hunk 2 above)
Comment (soupglasses) [reply README.md]: Changed it to (Advanced) to signify its non standard use.

File: README.md (hunk 1 above)
Comment (soupglasses) [reply README.md]: Oops! Its now fixed.

**Comment by soupglasses** [comment]
Updated a couple of things, notably adding a table of contents and renaming the files to be more clear of their behavior.

**Comment by soupglasses** [comment]
Painstakingly added build and install tests for each type of install we support and CI runs correctly.

Was a good idea to add, as i missed some fetchcontent + system-install differences with it.

**Comment by soupglasses** [comment]
See successful CI in: https://github.com/soupglasses/mtest/commit/af4cc731e774812de940132d874243c9ccbe4e90

**Pull request merged**